assessmentModel/src/commonTest/kotlin/org/sagebionetworks/assessmentmodel/AssessmentExtensionsTest.kt

```diff
@@ -0,0 +1,274 @@
+        for (column in columns) {
+            // Check that columnName matches the identifier of a node in the assessment
+            val node = children.first { it.resultId() == column.columnName } as Question
+            val result = node.createResult() as AnswerResult
+            // Check that the column type matches the baseType of the result
+            assertEquals(result.answerType, column.type)
+            assertEquals(node.title, column.questionTitle)
+        }
+    }
+
+    @Test
+    fun testToFlatAnswersJson() {
+        val serializer = PolymorphicSerializer(Assessment::class)
+        val assessment: Assessment = Serialization.JsonCoder.default.decodeFromString(serializer, assessmentJson)
+        val result = buildResults(assessment as AssessmentObject)
+        val answers = result.toFlatAnswers()
+        assertEquals(9, answers.size)
+        val columns = assessment.toFlatAnswersDefinition()
+        assertEquals(9, columns.size)
+        for (column in columns) {
+            // Verify that the keys to the answers match the column names in the definition
+            assertTrue(answers.containsKey(column.columnName))
+        }
+        val expectedAnswers = mapOf(
+            "time_xmx" to "10:15",
+            "duration_nwr" to "8.0",
+            "likert_scale_bjj" to "1",
+            "sliderQ_dfjwnf" to "1",
+            "numericQ_rqmrrj" to "1",
+            "yearQ_rwphsn" to "1",
+            "free_text_qfj" to "Test String",
+            "single_select_xnn" to "Test String",
+            "multiple_select_rqk" to "Choice 1,Choice 2"
+        )
+        assertEquals(expectedAnswers, answers)
+
+    }
+
+    private fun buildResults(assessmentObject: AssessmentObject) : AssessmentResult {
+        val result = assessmentObject.createResult()
+        for (child in assessmentObject.children) {
+            val childResult = child.createResult()
+            if (childResult is AnswerResult) {
+                childResult.jsonValue = when (childResult.answerType!!) {
+                    is AnswerType.Measurement -> JsonPrimitive(2.0)
+                    is AnswerType.Time -> JsonPrimitive("10:15")
+                    is AnswerType.DateTime -> JsonPrimitive("2020-01-21T12:00:00.000-03:00")
+                    is AnswerType.Duration -> JsonPrimitive(8.0)
+                    is AnswerType.INTEGER -> JsonPrimitive(1)
+                    is AnswerType.Decimal -> JsonPrimitive(1.0)
+                    is AnswerType.BOOLEAN -> JsonPrimitive(true)
+                    is AnswerType.STRING -> JsonPrimitive("Test String")
+                    is AnswerType.Array -> {
+                        if ((childResult.answerType as AnswerType.Array).baseType == BaseType.INTEGER) {
+                            JsonArray(listOf(JsonPrimitive(1), JsonPrimitive(2)))
+                        } else {
+                            JsonArray(listOf(JsonPrimitive("Choice 1"), JsonPrimitive("Choice 2")))
```

Comment:
These choices don't match the ones in the node.
```suggestion
                            JsonArray(listOf(JsonPrimitive("Choice A"), JsonPrimitive("Choice B")))
```